dali/operators/audio/nonsilence_op.h

```diff
@@ -0,0 +1,202 @@
+// Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_OPERATORS_AUDIO_NONSILENCE_OP_H_
+#define DALI_OPERATORS_AUDIO_NONSILENCE_OP_H_
+
+#include <cmath>
+#include <limits>
+#include <memory>
+#include <utility>
+#include <vector>
+#include "dali/core/convert.h"
+#include "dali/kernels/kernel_manager.h"
+#include "dali/kernels/signal/decibel/decibel_calculator.h"
+#include "dali/kernels/signal/moving_mean_square.h"
+#include "dali/pipeline/data/views.h"
+#include "dali/pipeline/operator/operator.h"
+
+namespace dali {
+namespace detail {
+
+const int kNumOutputs = 2;
+
+template<typename InputType>
+struct Args {
+  TensorView<StorageCPU, const InputType, 1> input;
+  float cutoff_db;
+  float reference_power;
+  bool reference_max;
+  int window_length;
+  int reset_interval;
+};
+
+
+template<typename T>
+T max(TensorView<StorageCPU, T, DynamicDimensions> tv) {
+  T max = std::numeric_limits<T>::lowest();
+  for (int i = 0; i < tv.num_elements(); i++) {
+    max = std::max(max, tv.data[i]);
+  }
+  return max;
+}
+
+
+/**
+ * @brief Performs leading and trailing thresholding.
+ *
+ * Returns index of a first value above the threshold in the buffer and
+ * the length up to the point, where last value above the threshold appears in the buffer.
+ *
+ * If the length == 0, begin == 0
+ *
+ * Example:
+ * buffer: [0, 0, 0, 0, 50, 50, 0, 0]
+ * cutoff: 20
+ * return: {4, 2}
+ *
+ * @return (begin_idx, length)
+ */
+template<typename T>
+std::pair<int, int> LeadTrailThresh(span<const T> buffer, T cutoff) {
+  assert(buffer.size() > 0);
+  int begin = -1;
+  int end = buffer.size();
+  while (begin < end && buffer[++begin] < cutoff);  // NOLINT
+  if (begin == end) return {0, 0};  // Rest is silence
+  while (buffer[--end] < cutoff);  // NOLINT
```

Comment:
just wondering, what is the linter complaining about here?

Reply:
I already complained about it in the previous PR and I can see it was not adjusted
https://github.com/NVIDIA/DALI/pull/1656#discussion_r366790919

There is a valid reason that LINT goes bonkers in this place.
It's easier to analyze code when it tries not to shave one line of code by creative use prefix increment and decrement.

Reply:
Changed to `{}`